### ISSUE TITLE
Reference error when using @Schema(name = ..) (swagger.core.v3)

### ISSUE THREAD

**Issue by stiancognite** [swagger-api/swagger-core#2531]
I have this SomethingDTO:

```
@Schema(name = "Something")
public class SomethingDTO {
  @Schema(description = "Unique ID")
  private String id;

  public SomethingDTO(Something something) {
    id = something.getId();
  }

  public String getId() {
    return id;
  }
}
```

And I set the return type of a resource method like this:
```
  @Operation(
      operationId = "getSomething",
      summary = "Gives you something",
      responses = {
          @ApiResponse(
              responseCode = "200",
              description = "The something.",
              content = @Content(
                  schema = @Schema(implementation = SomethingDTO.class)
              )
          )
      }
  )
```

The problem is that the generated spec for this resource methods return type looks like this:
```
                "responses": {
                    "200": {
                        "description": "The something.",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/SomethingDTO"
                                }
                            }
                        }
                    }
                }
```
i.e. it refers to `SomethingDTO`, which isn't defined in the spec because it's called `Something`.

**Comment by frantuma** [comment]
Thank for spotting and reporting these; it's fixed in #2533 and latest snapshot.

**Comment by stiancognite** [comment]
Thank you!:)
